lib/checkclass.cpp

```diff
@@ -562,7 +562,12 @@ void CheckClass::initializeVarList(const Function &func, std::list<const Functio
         }
 
         // Before a new statement there is "[{};()=[]" or ::
-        if (! Token::Match(ftok, "{|}|;|(|)|=|[|::"))
+        // We can also have a new statement after any operators or comparisons
+        if (! Token::Match(ftok, "{|}|;|(|)|=|[|::|%op%|%comp%"))
```

Comment:
this might work, not sure. maybe it's not handled by the match compiler yet. there used to be a requirement that the %..% patterns are put first. so I try to put them first always.